app/rtc_s1.py

```diff
@@ -561,13 +616,23 @@ def run(cfg):
     mosaic_kwargs['step_lon'] = dx
     _mosaic(output_imagery_list, geo_filename, interp='average',
             **mosaic_kwargs)
-    output_file_list.append(geo_filename)
+    nlooks_list = output_metadata_dict['nlooks'][1]
+    if mosaic_geobursts.check_mosaic_eligibility(output_imagery_list, nlooks_list):
+        mosaic_geobursts.weighted_mosaic(output_imagery_list, nlooks_list,
+                                         geo_filename.replace('.tif','.weighted_avg.tif'), cfg.geogrid)
+
+
+
+    output_file_list.append(geo_filename) #NOTE: This line will raise error when the used does not opt in to save auxiliary data (local incidence, simulated brightness, etc.)
 
     # mosaic other bands
     for key in output_metadata_dict.keys():
         output_file, input_files = output_metadata_dict[key]
         info_channel.log(f'mosaicking file: {output_file}')
         _mosaic(input_files, output_file, **mosaic_kwargs)
+        mosaic_geobursts.weighted_mosaic(input_files, nlooks_list,
+                                         output_file.replace('.tif','.weighted_avg.tif'),
```

Comment:
why you are changing the name of the output? The output_file is set in the runconfig. We should not use a different name.

Reply:
That was for the output comparison between `_moasic` and `mosaic_geobursts.weighted_mosaic` I've removed the part to modify the output.